### PR TITLE
Added formats(epub, pdf, htmlzip) to .readthedocs.yml

### PR DESCRIPTION
**Pull request by joelwilliam2005**
This pull request fixes issue #12484 "Configure Read The Docs to build all formats".

### CODE REVIEW

**Comment by lb-** [comment]
@joelwilliam2005  thank you, the change looks correct.

However, the RTD dev build is not showing the items

Any ideas @laymonage ? I think we can probably merge this and then see if we have any issues once it's deploying to `latest`.

I cannot see anything in the docs that say offline formats will be restricted though.

https://docs.readthedocs.io/en/stable/guides/enable-offline-formats.html

File: .readthedocs.yml
Comment (laymonage) [inline .readthedocs.yml:17]: Please simplify these changes, we don't need to specify the Sphinx configuration file, and we can just specify all formats.

```suggestion
formats: all
```

**Review by laymonage — CHANGES_REQUESTED** [review]
@lb- @joelwilliam2005 It turns out RTD doesn't build offline formats for PR builds. See https://docs.readthedocs.io/en/stable/config-file/v2.html#formats

> With [builds from pull requests](https://docs.readthedocs.io/en/stable/pull-requests.html), only HTML formats are generated. Other formats are resource intensive and will be built after merging.

**Review by lb- — APPROVED** [review]
Thanks @joelwilliam2005  and also @laymonage  for helping with that reference.

I will rebase, add a changelog and merge shortly.

**Comment by lb-** [comment]
Potential candidate for cherry pick to 6.4

I'll flag with the core team

**Comment by lb-** [comment]
Cherry picked to 6.3 f1ee2f5c73056095cdfb9b1a6c72f936619bbe8c & parents

**Pull request merged**
